types/datasource.d.ts

```diff
@@ -76,28 +86,194 @@ import {IsolationLevel, Transaction} from './transaction-mixin';
  */
 export declare class DataSource extends EventEmitter {
   name: string;
-  settings: Options;
+  settings: ConnectorSettings;
 
   initialized?: boolean;
   connected?: boolean;
   connecting?: boolean;
+  
+  /**
+   * {@inheritDoc DataSource.connector}
+   * @deprecated Use {@link DataSource.connector} instead.
+   */
+  adapter?: Connector;
 
+  /**
+   * Connector instance.
+   */
   connector?: Connector;
 
-  definitions: {[modelName: string]: ModelDefinition};
+  modelBuilder: ModelBuilder;
 
+  models: Record<string, typeof ModelBase>;
+  
+  definitions: {[modelName: string]: ModelDefinition};
+  
   DataAccessObject: AnyObject & {prototype: AnyObject};
+  
+  pendingConnectCallbacks?: Callback[];
+
+  /**
+   * Log benchmarked message.
+   * 
+   * @remarks
+   * This property is assigned to the defined to the attached connector's
+   * {@link Connector.log | log} class member.
+   * 
+   * @param sql 
+   * @param t Start time 
+   */
+  private log(sql: string, t: number): void;
+
+  private _queuedInvocations: number;
 
-  constructor(name: string, settings?: Options, modelBuilder?: ModelBuilder);
+  private _operations: Record<string, OperationOptions>;
 
-  constructor(settings?: Options, modelBuilder?: ModelBuilder);
+  operations(): Record<string, OperationOptions>;
+
+  defineOperation(name: string, options: OperationOptions, fn: Function): void;
+  
+  enableRemote(operation: string): void;
+
+  disableRemote(operation: string): void;
+
+  /**
+   * Default global maximum number of event listeners.
+   * 
+   * @remarks
+   * This default can be overriden through
+   * {@link ConnectorSettings.maxOfflineRequests}.
+   */
+  static DEFAULT_MAX_OFFLINE_REQUESTS: number;
+
+  /**
+   * A hash-map of the different relation types.
+   */
+  static relationTypes: Record<string, string>;
+
+  constructor(name: string, settings?: ConnectorSettings, modelBuilder?: ModelBuilder);
+
+  constructor(settings?: ConnectorSettings, modelBuilder?: ModelBuilder);
 
   constructor(
     connectorModule: Connector,
-    settings?: Options,
+    settings?: ConnectorSettings,
     modelBuilder?: ModelBuilder,
   );
 
+  private setup(dsName: string, settings: ConnectorSettings): void;
+  private setup(settings: ConnectorSettings): void;
+
+  private _setupConnector();
+
+  /**
+   * Get the maximum number of event listeners
+   * 
+   * @remarks
+   * Defaults to {@link DataSource.DEFAULT_MAX_OFFLINE_REQUESTS} if not explicitly
+   * configured in {@link ConnectorSettings.maxOfflineRequests}.
+   */
+  getMaxOfflineRequests(): number;
+
+  // Reason for deprecation is not clear.
+  /**
+   * {@inheritDoc Connector.getTypes}
+   * @deprecated
+   */
+  getTypes(): string[];
+
+  /**
+   * Check if the datasource supports the specified types.
+   * @param types Type name(s) to check against
+   */
+  supportTypes(types: string | string[]): boolean;
```

Comment:
Based on the original codebase comments, `getTypes()` is deprecated but not `supportTypes()`; Not sure of the reason for this. The original purpose of these functions are also not clear.